presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -144,6 +148,30 @@ public void testNumeric()
         assertQueryUsingH2Cte("SELECT linenumber, ship_by_air, is_returned FROM lineitem_ex WHERE orderkey < 30000 AND ship_by_air = true");
     }
 
+    @Test
+    public void testTimestamps()
+    {
+        // Single timestamp column
+        assertQueryUsingH2Cte("SELECT ship_timestamp FROM lineitem_ex");
+
+        assertQueryUsingH2Cte("SELECT ship_timestamp FROM lineitem_ex WHERE ship_timestamp < TIMESTAMP '1993-01-01 01:00:00'");
+
+        assertQueryUsingH2Cte("SELECT count(*) FROM lineitem_ex WHERE ship_timestamp IS NOT NULL");
+
+        assertQueryUsingH2Cte("SELECT count(*) FROM lineitem_ex WHERE ship_timestamp = (TIMESTAMP '2012-08-08 01:00:00')");
+
+        // Two timestamp columns
+        assertQueryUsingH2Cte("SELECT commit_timestamp, ship_timestamp FROM lineitem_ex");
+
+        assertQueryUsingH2Cte("SELECT commit_timestamp, ship_timestamp FROM lineitem_ex WHERE ship_timestamp > (TIMESTAMP '1993-08-08 01:00:00') AND commit_timestamp < (TIMESTAMP '1993-08-08 01:00:00')");
+
+        assertQueryUsingH2Cte("SELECT commit_timestamp, ship_timestamp FROM lineitem_ex WHERE year(ship_timestamp) - year(commit_timestamp) > 1");
+
+        assertQueryUsingH2Cte("SELECT COUNT(*) FROM lineitem_ex WHERE commit_timestamp IS NULL");
```

Comment:
this is essentially the same query as in L159; is there any particular reason for duplication?

Reply:
will remove it